pkg/controller/controller.go

```diff
@@ -114,6 +114,18 @@ const (
 	MigrationBackoffReason = "MigrationBackoff"
 )
 
+type PodCacheStore struct {
+	indexer cache.Indexer
+}
+
+func NewPodCacheStore(indexer cache.Indexer) *PodCacheStore {
+	return &PodCacheStore{indexer: indexer}
+}
+
+func (p *PodCacheStore) CurrentPod(vmi *v1.VirtualMachineInstance) (*k8sv1.Pod, error) {
+	return CurrentVMIPod(vmi, p.indexer)
+}
```

Comment:
What value does this bring?

Reply:
It allows the code that uses it to be testable. See the use of a stub to replace it in the tests below (`stubPodGetter`).

Reply:
Imho you could just keep the informer and use the `CurrentVMIPod` directly. Creating the `Store` for the tests as you need. This seems like additional complexity/boilerplate.

Reply:
The `CurrentVMIPod` has logic that I do not want to depend on.
I do not want the network tests to know what an informer is, how its data is stored and so on.

For tests to be maintainable, they need to be simple and depend as little as possible on other logic.

The logic of `CurrentVMIPod` should be unit tested directly and there I guess an informer sore will be a good fit.

Reply:
Sorry meant to say "You could just keep the `Store` from informer...."

>For tests to be maintainable, they need to be simple and depend as little as possible on other logic.

In the case I described you would just pass relevant Store to the test so it would be simple, imho

>The logic of CurrentVMIPod should be unit tested directly and there I guess an informer store will be a good fit.

100% and funny enough it is not :)

Reply:
> In the case I described you would just pass relevant Store to the test so it would be simple, imho

I do not thin it is simple. But even if it is, it assumes I know the logic if that method and I do not want to know how it works at all. I actually think it is wrong to depend on knowing how it works.